src/main/java/net/imglib2/parallel/Parallelization.java

```diff
@@ -0,0 +1,266 @@
+		{
+			runWithExecutor( taskExecutor, action );
+		}
+	}
+
+	/**
+	 * To run an algorithm a given number of threads use:
+	 * <p>
+	 * {@code output = Parallelization.runWithNumThreads( numThreads, () -> myAlgorithm( input ) );}
+	 */
+	public static < R > R runWithNumThreads( int numThreads, Callable< R > action )
+	{
+		try (TaskExecutor taskExecutor = TaskExecutors.numThreads( numThreads ))
+		{
+			return runWithExecutor( taskExecutor, action );
+		}
+	}
+
+	/**
+	 * Executes the given {@link Runnable} with the given {@link ExecutorService},
+	 * and waits for the execution to finish.
+	 */
+	public static void runWithExecutor( ExecutorService executorService, Runnable action )
+	{
+		runWithExecutor( TaskExecutors.forExecutorService( executorService ), action );
+	}
+
+	/**
+	 * Executes the given {@link Callable} with the given {@link ExecutorService},
+	 * waits for the execution to finish and returns the result.
+	 */
+	public static < R > R runWithExecutor( ExecutorService executorService, Callable< R > action )
+	{
+		return runWithExecutor( TaskExecutors.forExecutorService( executorService ), action );
+	}
+
+	/**
+	 * Executes the given {@link Runnable} with the given {@link TaskExecutor},
+	 * and waits for the execution to finish.
+	 */
+	public static void runWithExecutor( TaskExecutor taskExecutor, Runnable action )
+	{
+		try (Frame frame = setExecutorRequiresReset( taskExecutor ))
+		{
+			action.run();
+		}
+	}
+
+	/**
+	 * Executes the given {@link Callable} with the given {@link TaskExecutor},
+	 * waits for the execution to finish and returns the result.
+	 */
+	public static < T > T runWithExecutor( TaskExecutor taskExecutor, Callable< T > action )
+	{
+		try (Frame frame = setExecutorRequiresReset( taskExecutor ))
+		{
+			return action.call();
+		}
+		catch ( Exception e )
+		{
+			throw new RuntimeException( e );
+		}
+	}
+
+	/**
+	 * This method can be used to execute an algorithm with a given {@link TaskExecutor}.
+	 * But it's easier to use {@link #runWithExecutor}.
+	 * <p>
+	 * This method sets the {@link TaskExecutor} for the current thread.
+	 * This can be used to execute a certain part of your code with the given
+	 * {@link TaskExecutor}. It's mandatory to call the {@code close()} of the
+	 * return {@link Frame frame} afterwards. This could be done using an
+	 * try-with-resources statement, that was introduced with Java 7.
+	 * <pre>
+	 * {@code
+	 *
+	 * try ( Parallelization.Frame frame = Parallelization.setExecutorRequiresReset( taskExecutor ) )
+	 * {
+	 *     myAlgorithm(input);
+	 * }
+	 * }
+	 * </pre>
+	 * Or by explicitly calling {@code frame.close()} in the finally block.
+	 * <pre>
+	 * {@code
+	 *
+	 * Parallelization.Frame frame = Parallelization.setExecutorRequiresReset( taskExecutor );
+	 * try
+	 * {
+	 *     myAlgorithm(input);
+	 * }
+	 * finally
+	 * {
+	 *     frame.close();
+	 * }
+	 * }
+	 * </pre>
+	 */
+	public static Frame setExecutorRequiresReset( TaskExecutor taskExecutor )
```

Comment:
I'm not sure this is needed/should be public.
The example in the javadoc could be just `runWithExecutor(taskExecutor, () -> myAlrogithm(input));`?
Where is this useful?

Reply:
This is useful, if wrapping the method call into a lambda is not an option. We could for example write a `@PreprocesserPlugin` and a `@PostprocessorPlugin` for SciJava. The  `@PreprocesserPlugin` plugin would call this method, to set the `TaskExecutor` for a `Command`, and the `@PostprocessorPlugin` would close the frame.

But the are other options too, the `Command` could use `TaskExector.setExecutor(...)`, or maybe the `ModuleService` would do so. @ctrueden What are your thought's on this?

Reply:
My initial thought is that using `runWithExecutor` in the `ModuleRunner` class, which governs the module execution process, would be sufficient for configuring SciJava module execution appropriately. For now, my vote is to make `setExecutorRequiresReset` private, but leave it otherwise exactly the same including the javadoc, since it is very helpful to understand what it is for. It would be fine to later make it public, should we encounter a case where that is needed, whereas making it private later is a breaking API change.